JsonSchema/Serialization/ValidatingJsonConverter.cs

```diff
@@ -112,6 +112,7 @@ public ValidatingJsonConverter(JsonSchema schema, Func<JsonSerializerOptions, Js
 		var newOptions = _optionsFactory(options);
 
 		if (validation.IsValid)
+			// TODO: this does not use the options reader because at least one test failed with a NotSupportedException
```

Comment:
I'd like this explored, please.

Reply:
The specific message:

> System.NotSupportedException: 'Metadata for type 'Json.Schema.Tests.Serialization.DeserializationTests+FooWithSchema' was not provided to the serializer. The serializer method used does not support reflection-based creation of serialization-related type metadata. If using source generation, ensure that all root types passed to the serializer have been indicated with 'JsonSerializableAttribute', along with any types that might be serialized polymorphically.'

It sounds like the source generators aren't picking up the type transitively; I'm not familiar enough with how those work to solve it at this time.

Reply:
Okay.  I'll explore this a bit.